metrics-operator/converter/sli_converter_test.go

```diff
@@ -0,0 +1,163 @@
+package converter
+
+import (
+	"testing"
+
+	metricsapi "github.com/keptn/lifecycle-toolkit/metrics-operator/api/v1alpha3"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const sliContent = `
+spec_version: "1.0"
+indicators:
+  throughput: "builtin:service.requestCount.total:merge(0):count?scope=tag(keptn_project:$PROJECT),tag(keptn_stage:$STAGE),tag(keptn_service:$SERVICE),tag(keptn_deployment:$DEPLOYMENT)"
+  response_time_p95: "builtin:service.response.time:merge(0):percentile(95)?scope=tag(keptn_project:$PROJECT),tag(keptn_stage:$STAGE),tag(keptn_service:$SERVICE),tag(keptn_deployment:$DEPLOYMENT)"`
+
+const expectedOutput = `---
+apiVersion: metrics.keptn.sh/v1alpha3
+kind: AnalysisValueTemplate
+metadata:
+  creationTimestamp: null
+  name: response_time_p95
+spec:
+  provider:
+    name: dynatrace
+    namespace: keptn
+  query: builtin:service.response.time:merge(0):percentile(95)?scope=tag(keptn_project:{{.project}}),tag(keptn_stage:{{.stage}}),tag(keptn_service:{{.service}}),tag(keptn_deployment:{{.deployment}})
+---
+apiVersion: metrics.keptn.sh/v1alpha3
+kind: AnalysisValueTemplate
+metadata:
+  creationTimestamp: null
+  name: throughput
+spec:
+  provider:
+    name: dynatrace
+    namespace: keptn
+  query: builtin:service.requestCount.total:merge(0):count?scope=tag(keptn_project:{{.project}}),tag(keptn_stage:{{.stage}}),tag(keptn_service:{{.service}}),tag(keptn_deployment:{{.deployment}})
```

Comment:
shpould these be capitalized? eg {{.Project}}

Reply:
if we assume we are going to substitute these strings from Analysis CR, which content will look like:

```
project: project1
stage: stage1
```

The keys are not capitalized, as k8s does not allow capitalized keys from it's definition, so it's better (easier) for us not to have them capitalized